pkgs/test/test/runner/line_and_col_test.dart

```diff
@@ -80,6 +110,52 @@ void main() {
       await test.shouldExit(0);
     });
 
+    test('additionally selects groups with a matching custom location',
+        () async {
+      // TODO(dantup): This fails because we don't ever look at a groups
+      //  location.. instead, we only look at tests. The reason we can
+      //  normally run groups by line/col is because they just happen to be
+      //  in the stack trace for the test() call too.
+      //
+      //  So maybe we need to look up the tree to match location (in
+      //  Runner._loadSuites() filter)?
```

Comment:
Currently this test fails, because we only put the `TestLocation` on the `group`. It turns out that all filtering is just done based on tests, and the reason we can run a test by the groups line+col is because that location also happens to be in the call stack of the `test()` call (and therefore we match on the test, not the group).

So here, we fail to match anything, because the tests `location`s are `null` and don't match, and we never look at the group.

My feeling is that we need to change the filter to also look up the chain at parents when checking the `location`, however at a glance we don't seem to have easy access to that in the place where we're filtering (`Runner._loadSuites`).

Reply:
Ah yeah that makes sense, I do think we will end up needing to pass through the parent groups then.

Reply:
hmm, I'm struggling to see how best to do this - it seems we don't have a reference from a `Group`/`GroupEntry` to the parent. Any pointers?

Reply:
In the Declarer, the `Group` entries are built bottom-up.

You could add a `Group? parent;` field to `GroupEntry`, and then assign it inside `Declarer.build` - setting the "parent" of each `GroupEntry` in `_entries` to the group you just created.

You could similarly assign the parent groups when deserializing tests over the wire this way (after creating the group inside `_Deserializer.deserializeGroup`).

I really don't know all that much about this part of the codebase though, so this approach could run into additional snags that I am not expecting.

Reply:
Thanks! I did this and it seems to have work. I wired up the parents in both the declarer and deserialization, although I'm not sure if both are necessary (it felt better than it always being null on one side though?).

I'm not sure having a mutable `parent` is ideal, but I'm not sure if that's avoidable (we can't pass parent to child and child to parent in constructors immutably?).